.eslintrc.js

```diff
@@ -55,6 +53,12 @@ module.exports = {
       files: [ '**/*.ts' ],
 
       parser: '@typescript-eslint/parser',
+      plugins: ['@typescript-eslint'],
+      extends: [
+        'plugin:@typescript-eslint/eslint-recommended',
```

Comment:
[`plugin:@typescript-eslint/eslint-recommended`](https://github.com/typescript-eslint/typescript-eslint/blob/v2.33.0/packages/eslint-plugin/src/configs/eslint-recommended.ts) turns off all the `eslint` rules that conflict with `@typescript-eslint`

Reply:
Do these arrays get merged with the outer config, or do they override it completely?

Reply:
I believe they get merged, in the order they are defined. Though the specifics are pretty vague..

- Mentioned briefly in the [eslint migration guide](https://eslint.org/docs/user-guide/migrating-to-7.0.0#lint-files-matched-by-overridesfiles-by-default)
- https://github.com/eslint/rfcs/blob/master/designs/2019-additional-lint-targets/README.md

You can run `npx eslint --print-config ./packages/@ember/string/index.ts` to see what the eslint config for that file will look like..

```json
{
  "env": {},
  "globals": {},
  "parser": "./node_modules/@typescript-eslint/parser/dist/parser.js",
  "parserOptions": {
    "sourceType": "module",
    "project": "./tsconfig.json",
    "tsconfigRootDir": "."
  },
  "plugins": [
    "prettier",
    "disable-features",
    "qunit",
    "import",
    "ember-internal",
    "@typescript-eslint"
  ],
  "rules": {
    ...
  },
  "settings": {
    ...
  },
  "ignorePatterns": [
    ...
  ]
}
```

..compared with the config for a `.js` file `npx eslint --print-config packages/@ember/application/index.js`

```json
{
  "env": {},
  "globals": {
    "window": true,
    "document": true,
    "setTimeout": true,
    "clearTimeout": true,
    "setInterval": true,
    "clearInterval": true,
    "console": true,
    "Map": true,
    "Set": true,
    "Symbol": true,
    "WeakMap": true
  },
  "parser": "./node_modules/babel-eslint/lib/index.js",
  "parserOptions": {
    "ecmaVersion": 2017,
    "sourceType": "module"
  },
  "plugins": [
    "prettier",
    "disable-features",
    "qunit",
    "import",
    "ember-internal"
  ],
  "rules": {
    ...
  },
  "settings": {
    ...
  },
  "ignorePatterns": [
    ...
  ]
}
```

Reply:
Looking at this now.. the overrides for `files: [ 'packages/**/*.js' ]` may need to be updated to include `packages/**/*.ts`?

Reply:
I've removed the `@typescript-eslint` changes as #19166 made it difficult to keep them, without a lot of changes. I can reintroduce it in a different PR